### PR TITLE
Update RELEASE.md

### PR DESCRIPTION
**Pull request by xtreme-sameer-vohra**
## Proposed Changes
- Update releaseability steps to capture Module Release
- Minor typo fix for knative version vs. release version
- Minor sentence changes

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -26,19 +28,18 @@ workflow in [release.yaml](.github/workflows/release.yaml).
 work. Here's one example workflow (assuming of course you have rights to
 directly push to the upstream repo).
 
-Create a temporary clone of the repo, e.g. to `/tmp`. Then create the `tag` and
-push to origin. Despite Knative 1.0, the release tags in this repo are still versioned in the `v0.x.y` format and correspond to the Knative tags `knative-v1.x.y`.
-This starts from `v0.27.0` == `knative-v1.0.0` and each MINOR version addition in the `knative-v1.x.y` tag corresponds to the same addition on the `0.x.z` tag.
+Despite Knative's external semantic version uses a MAJOR version like 1.0, the release tags in this repo are still versioned in the `v0.x.y` format and correspond to the Knative tags `knative-v1.a.y`.
```
[the remark this answers is not in the document]
Comment (dprotaso) [reply RELEASE.md:31]: yeah the v0.x.y versioning is to signal we don't want to commit to strict semver versioning for our go packages.

I'd leave it up to you all here to decide what's best for this repo

[the remark this answers is not in the document]
Comment (gabo1208) [reply RELEASE.md:31]: I think to maintain the knative feel to it it's ok. And you are right ébano let me change those variables not to overlap

[the remark this answers is not in the document]
Comment (xtreme-sameer-vohra) [reply RELEASE.md:31]: Some more context https://docs.google.com/document/d/1_khEw5_l8Zhh4FuY3NM56o_OM-cNWmmq1vXfLCxZqMA/edit#heading=h.j9wh96sz9rvq
AFAIK, we need the `v0.x.y` tags in this repo to continue using [the existing automation](https://github.com/vmware-tanzu/sources-for-knative/actions/workflows/release.yaml) and having the `knative-1.a.y` tags for downstream to match releases with Knative OSS. 

@gabo1208 is that correct ?

[the remark this answers is not in the document]
Comment (embano1) [reply RELEASE.md:31]: Thx for the clarification, I understand it now 🙏

[the remark this answers is not in the document]
Comment (gabo1208) [reply RELEASE.md:31]: Exactly that, thanks @xtreme-sameer-vohra !

**Review by gabo1208 — APPROVED** [review]

**Pull request merged**